### PR TITLE
Planner-1234 Conference scheduling crowd control constraint

### PR DESCRIPTION
**Pull request by MusaTalluzi**

### CODE REVIEW

File: optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/domain/Talk.java [hunk 1]
```diff
@@ -228,8 +230,10 @@ public int overlappingMutuallyExclusiveTalksTagCount(Talk other) {
         return (int) mutuallyExclusiveTalksTagSet.stream().filter(tag -> other.mutuallyExclusiveTalksTagSet.contains(tag)).count();
     }
 
-    public boolean hasPrerequisite(Talk talk) {
-        return prerequisiteTalkSet.contains(talk);
+    public int missingPrerequisiteCount() {
+        return (int) prerequisiteTalkSet.stream()
+                .filter(prerequisite -> prerequisite.getTimeslot() == null || timeslot.endsBefore(prerequisite.getTimeslot()))
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/java/org/optaplanner/examples/conferencescheduling/domain/Talk.java]: Good one that you fixed this. This affects the Construction Heuristics's intializingTrend.
Basically, it's better if by initializing more Talks (picking a room/timeslot for each one), the score only goes down, never up.

File: optaplanner-examples/src/main/resources/org/optaplanner/examples/conferencescheduling/solver/conferenceSchedulingScoreRules.drl [hunk 2]
```diff
@@ -358,4 +357,25 @@ rule "Same day talks"
     then
         scoreHolder.addSoftConstraintMatch(kcontext, - $weight * ($rightTalk.overlappingThemeTrackCount($leftTalk)
             + $rightTalk.overlappingContentCount($leftTalk)));
+end
+
+rule "Popular talks"
```
Comment (ge0ffrey) [inline optaplanner-examples/src/main/resources/org/optaplanner/examples/conferencescheduling/solver/conferenceSchedulingScoreRules.drl]: There's 2 ways to formalize the constraint that Stephan was talking about:

- For 2 talks in the same timeslot, the more popular talk needs to be in the bigger room.
- For 2 talks, the more popular talk needs to be in the bigger room. That's what currently is implemented and is probably best. Once we do non-disruptive re-planning, the other hard constrainst will turn that into the first form, but during the initial schedule this ways is indeed better.

I 'd like the name of the constraint to reflect our choice.

Comment (MusaTalluzi) [reply optaplanner-examples/src/main/resources/org/optaplanner/examples/conferencescheduling/solver/conferenceSchedulingScoreRules.drl]: @ge0ffrey what other hard constraint you're referring to?

**Comment by ge0ffrey** [comment]
OrderByMoveIndexBlockingQueueTest.addExceptionThrown was failing again

**Pull request merged**